components/button/button-subtle.js

```diff
@@ -213,6 +220,19 @@ class ButtonSubtle extends ButtonMixin(RtlMixin(LitElement)) {
 		`;
 	}
 
+	willUpdate(changedProperties) {
+		if (changedProperties.has('_hasCustomIcon') || changedProperties.has('icon')) {
+			this._hasIcon = this._hasCustomIcon || this.icon;
+		}
+	}
+
+	_handleIconSlotChange(e) {
+		const icon = e && e.target && e.target.assignedNodes({ flatten: true }).filter((node) => {
```

Comment:
Maybe one day we will be able to just use CSS  with the`:hasSlotted(d2l-icon-custom)` or `:has-assigned(d2l-icon-custom)` that's been proposd. https://github.com/w3c/csswg-drafts/issues/6867 😀